### PR TITLE
ODIM-5349 Update odimra config with message bus details 

### PR DESCRIPTION
**Pull request by irappa-pattar**

### CODE REVIEW

**Review by jeevan-kamkar — APPROVED** [review]

File: odim-controller/scripts/kube_deploy_nodes.yaml.tmpl [hunk 1]
```diff
@@ -58,6 +58,9 @@ odimra:
   apiNodePort: 30080
   kafkaNodePort: 30092
   
+  messageBusType:
+  messageBusQueue:
+
```
Comment (amar-shalgar) [inline odim-controller/scripts/kube_deploy_nodes.yaml.tmpl:63]: Add default values here

File: svc-task/main.go [hunk 2]
```diff
@@ -43,10 +44,11 @@ func main() {
 
 	config.CollectCLArgs()
 
-	if err := dc.SetConfiguration(config.Data.MessageQueueConfigFilePath); err != nil {
-		log.Fatal("error while trying to set messagebus configuration: " + err.Error())
+	if strings.EqualFold(config.Data.MessageBusConf.MessageBusType, "kafka") {
+		if err := dc.SetConfiguration(config.Data.MessageBusConf.MessageQueueConfigFilePath); err != nil {
+			log.Fatal("error while trying to set messagebus configuration: " + err.Error())
+		}
```
Comment (amar-shalgar) [inline svc-task/main.go:50]: I can see only kafka condition, similarly add code for redisstreams

File: lib-utilities/config/constants.go [hunk 3]
```diff
@@ -142,3 +142,9 @@ var (
 		"TLS_1.2": tls.VersionTLS12,
 	}
 )
+
+// AllowedMessageBusTypes is for checking for message types are allowed
+var AllowedMessageBusTypes = map[string]bool{
+	"kafka":       true,
+	"RedisStream": true,
+}
```
Comment (amar-shalgar) [inline lib-utilities/config/constants.go:150]: cases should match across the options, keep it either small/caps/camelcase for both options

**Review by amar-shalgar — CHANGES_REQUESTED** [review]
Please address changes

File: odim-controller/scripts/kube_deploy_nodes.yaml.tmpl (hunk 1 above)
Comment (irappa-pattar) [reply odim-controller/scripts/kube_deploy_nodes.yaml.tmpl:63]: added

File: lib-utilities/config/constants.go (hunk 3 above)
Comment (irappa-pattar) [reply lib-utilities/config/constants.go:150]: updated

File: svc-task/main.go (hunk 2 above)
Comment (irappa-pattar) [reply svc-task/main.go:50]: we are setting configuration for Kafka, there is no such config method for RedisStream in this PR

**Review by amar-shalgar — APPROVED** [review]
LGTM

**Review by shivacharanms — APPROVED** [review]

**Pull request merged**